File: apps/mobile-native/ios/NexaraApp/UI/Common/OfflineBanner.swift
import SwiftUI

/// Banner cuando no hay red o hay mutaciones pendientes.
struct OfflineBanner: View {
    @ObservedObject private var network = NetworkMonitor.shared
    @State private var pending = 0
    @State private var syncing = false

    var body: some View {
        Group {
            if !network.isOnline {
                banner(
                    pending > 0
                        ? "Sin conexión — \(pending) en cola (incl. fotos)"
                        : "Sin conexión — cambios y fotos se encolan localmente",
                    color: .orange,
                    tappable: false
                )
            } else if pending > 0 {
                banner(
                    syncing
                        ? "Sincronizando \(pending)…"
                        : "Pendientes de sync: \(pending) · toca para reintentar",
                    color: .blue,
                    tappable: !syncing
                )
            }
        }
        .task {
            pending = OfflineMutationQueue.shared.pendingCount
        }
        .onReceive(NotificationCenter.default.publisher(for: .nexaraOfflineQueueChanged)) { _ in
            pending = OfflineMutationQueue.shared.pendingCount
        }
    }

    private func banner(_ text: String, color: Color, tappable: Bool) -> some View {
        Text(text)
            .font(.caption)
            .foregroundColor(.white)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 6)
            .background(color)
            .onTapGesture {
                guard tappable else { return }
                Task {
                    syncing = true
                    await OfflineSyncCoordinator.shared.replay()
                    pending = OfflineMutationQueue.shared.pendingCount
                    syncing = false
                }
            }
    }
}

extension Notification.Name {
    static let nexaraOfflineQueueChanged = Notification.Name("nexara.offline.queue.changed")
}

File: apps/mobile-native/android/app/src/main/java/mx/nexara/mobile/nativeapp/ui/shared/OfflineBanner.kt
package mx.nexara.mobile.nativeapp.ui.shared

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun OfflineBanner(
    isOffline: Boolean,
    pendingMutations: Int,
    onSyncNow: (() -> Unit)? = null,
) {
    val (bg, msg) = when {
        isOffline && pendingMutations > 0 ->
            Color(0xFFB45309) to "Sin conexión · $pendingMutations en cola (incl. fotos)"
        isOffline ->
            Color(0xFFB45309) to "Sin conexión · cambios y fotos se encolan localmente"
        pendingMutations > 0 ->
            Color(0xFF0369A1) to "Pendientes de sync: $pendingMutations · toca para reintentar"
        else -> return
    }
    Text(
        text = msg,
        modifier = Modifier
            .fillMaxWidth()
            .background(bg)
            .then(
                if (!isOffline && pendingMutations > 0 && onSyncNow != null) {
                    Modifier.clickable { onSyncNow() }
                } else Modifier,
            )
            .padding(horizontal = 16.dp, vertical = 8.dp),
        color = Color.White,
        style = MaterialTheme.typography.labelMedium,
        fontWeight = FontWeight.SemiBold,
    )
}
